crashing with UnboundLocalError in create_filter_process

```
        try:
            yield stream
    
        except Exception:
            # something went wrong with processing the stream by borg
            logger.debug('Exception, killing the filter...')
>           proc.kill()
E           UnboundLocalError: local variable 'proc' referenced before assignment

src/borg/helpers/process.py:335: UnboundLocalError
```

Guess cmd can be None and there should be a `if cmd:`.

--- a/src/borg/helpers/process.py
+++ b/src/borg/helpers/process.py
@@ -332,7 +332,8 @@ def create_filter_process(cmd, stream, stream_close, inbound=True):
     except Exception:
         # something went wrong with processing the stream by borg
         logger.debug('Exception, killing the filter...')
-        proc.kill()
+        if cmd:
+            proc.kill()
         borg_succeeded = False
         raise
     else:

